shared/common-adapters/banner.js

```diff
@@ -0,0 +1,112 @@
+// @flow
+import * as React from 'react'
+import {Box2} from './box'
+import Icon, {castPlatformStyles as iconCastPlatformStyles} from './icon'
+import Text from './text'
+import * as Styles from '../styles'
+import {memoize} from '../util/memoize'
+
+type Color = 'blue' | 'red' | 'yellow' | 'green' | 'grey'
+
+type Props = {
+  actions?: Array<{
+    title: string,
+    onClick: () => void,
+  }>,
+  color: Color,
+  onClose?: () => void,
+  text: string,
+}
+
+const getContainerStyle = memoize((props: Props) =>
```

Comment:
we shouldn't memoize common adapters like this. you'll have no control over your cache getting thrashed

Reply:
if you want to do this (not that i think this is expensive so we should actually not do this at all) but if you did just make a map of the color to the value and be done

Reply:
and in general this is not a good idea to pass props to memoize, esp if you just pulling out one value like color